code/modules/clothing/suits/armor.dm

```diff
@@ -1,6 +1,6 @@
 
 /obj/item/clothing/suit/armor
-	allowed = list(/obj/item/weapon/gun/energy,/obj/item/device/radio,/obj/item/weapon/reagent_containers/spray/pepper,/obj/item/weapon/gun/projectile,/obj/item/ammo_magazine,/obj/item/ammo_casing,/obj/item/weapon/melee/baton,/obj/item/weapon/handcuffs)
+	extra_allowed = list(/obj/item/device/radio,/obj/item/weapon/reagent_containers/spray/pepper)
```

Comment:
instead of that last entry, add
 /obj/item/weapon/reagent_containers/spray

to the base allowed list